### PR TITLE
Support for PUT requests.

### PR DESCRIPTION
**Pull request by IonicaBizau**
Plex uses PUT requests to update the data.

Fixes #9.

### CODE REVIEW

**Comment by IonicaBizau** [comment]
This is will update the video description of the video with the id `123`. It will change it into `foo`.

```js
client.postQuery("/library/sections/3/all?id=123&summary.value=foo")
```

**Comment by OverloadUT** [comment]
This would most definitely be breaking. `postQuery` should be `POST`, because it's called `postQuery`. If anything, `putQuery` should be added as a new method if PUT requests are indeed used by the Plex API.

**Comment by IonicaBizau** [comment]
@OverloadUT That's right. Addressed in 4acd4f5a988275ca9ea416ae8e924da3ee966e05.

File: lib/api.js [hunk 1]
```diff
@@ -87,6 +87,21 @@ PlexAPI.prototype.postQuery = function query(options) {
     return this._request(options).then(uri.attach(url));
 };
 
+PlexAPI.prototype.putQuery = function query(options) {
```
Comment (phillipj) [inline lib/api.js]: `function query(..)` should be `function putQuery(..)` here. It has impact on the potential stack traces generated.

Comment (phillipj) [inline lib/api.js]: ..see we've got the same mistake in the `postQuery()` function above. Could you please fix that one as well?

Comment (IonicaBizau) [inline lib/api.js]: @phillipj Sure!

**Comment by phillipj** [comment]
This looks like a very good start @IonicaBizau. Other than my inline comments, could you add some tests and document this function in `Readme.md`?

File: lib/api.js (hunk 1 above)
Comment (IonicaBizau) [inline lib/api.js]: @phillipj I thought it's intended. :grin: Fixed.

**Comment by IonicaBizau** [comment]
@phillipj I'm currently busy, but will take care of it in the next few days hopefully. Ping me if I forget.

**Comment by phillipj** [comment]
No stress! Going on a short vacation myself, so I won't sit and wait 😃

**Comment by IonicaBizau** [comment]
@phillipj I just pushed the tests for `putQuery` method. Hope it's fine. :grin:

**Comment by IonicaBizau** [comment]
@phillipj :bell:

File: test/put-test.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		return api.putQuery(ROOT_URL);
+	});
+
+	it('promise should succeed when request response status code is 201', function() {
+		server.stop();
+		server.expectsPut({statusCode: 201});
+		return api.putQuery(ROOT_URL);
+	});
+
+	it('should result in a POST request', function() {
```
Comment (phillipj) [inline test/put-test.js]: Replace POST with PUT

Comment (IonicaBizau) [inline test/put-test.js]: @phillipj Done :grin:

**Comment by phillipj** [comment]
@IonicaBizau sorry, did see your comments until now.

The test describe nitpick is the only thing I see, the rest serms mergeable

**Comment by phillipj** [comment]
Hohoho lightning quick response there 😃

**Comment by IonicaBizau** [comment]
:tada: :rocket:

**Comment by IonicaBizau** [comment]
@phillipj Don't forget to `npm publish` a new version. Thanks! :grin:

**Comment by phillipj** [comment]
Thanks for your contributions! I'll make a new minor release asap

**Comment by phillipj** [comment]
@IonicaBizau v3.5.0 just released 🎆

**Pull request merged**